### PR TITLE
Add authorization and authentication with devise

### PR DESCRIPTION
**Pull request by jkoren**
We added 3 fields username, last name and first name to the schema, and added the ability to enter those fields in the new user sign up page (under registrations: new.html.erb).  We changed factory.bot to add those fields, and modified the test to fill those fields in.

### CODE REVIEW

File: db/migrate/20201020184453_add_username_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUsernameToUser < ActiveRecord::Migration[5.2]
+  def change
+        add_column :users, :username, :string, null: false
+  end
```
Comment (safang99) [inline db/migrate/20201020184453_add_username_to_user.rb:4]: Looks like line 3 may have been indented twice.  Let's shift that back to the left so its indented only 2 spaces within line 2

File: spec/features/user_signs_up_spec.rb [hunk 2]
```diff
@@ -18,6 +18,10 @@
     fill_in 'Email', with: 'john@example.com'
     fill_in 'Password', with: 'password'
     fill_in 'Password confirmation', with: 'password'
+    fill_in 'Username', with: 'john'
+    fill_in 'First name', with: 'John'
+    fill_in 'Last name', with: 'Smith'
+
```
Comment (safang99) [inline spec/features/user_signs_up_spec.rb]: Lets remove one of the extra empty lines here.  Either 24 or 25

File: spec/support/factory_bot.rb [hunk 3]
```diff
@@ -3,6 +3,9 @@
 FactoryBot.define do
   factory :user do
     sequence(:email) {|n| "user#{n}@example.com" }
+    first_name { 'john' }
+    last_name { 'smith' }
+    username { 'jsmitty' }
     password { 'password' }
     password_confirmation { 'password' }
   end
```
Comment (safang99) [inline spec/support/factory_bot.rb:11]: Nice!

**Review by safang99 — CHANGES_REQUESTED** [review]
Great job! 2 Super small formatting changes requested

File: db/migrate/20201020184453_add_username_to_user.rb (hunk 1 above)
Comment (jkoren) [reply db/migrate/20201020184453_add_username_to_user.rb:4]: We made this change.

File: spec/features/user_signs_up_spec.rb (hunk 2 above)
Comment (jkoren) [reply spec/features/user_signs_up_spec.rb]: We made this change.

File: app/views/devise/registrations/new.html.erb [hunk 4]
```diff
@@ -2,6 +2,20 @@
 
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
   <%= render "devise/shared/error_messages", resource: resource %>
+  <div class="field">
+    <%= f.label :first_name %><br />
+    <%= f.text_field :first_name, autofocus: true, autocomplete: "first name" %>
+  </div>
```
Comment (dovakeen118) [inline app/views/devise/registrations/new.html.erb:8]: `autofocus` can only be used for one input field in a form - it makes that field active and ready for input from the user to start typing

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Looks good - just need to remove the extra `autofocus: true` attributes from the new registration form!

File: app/views/devise/registrations/new.html.erb (hunk 4 above)
Comment (jkoren) [reply app/views/devise/registrations/new.html.erb:8]: I removed autofocus from the 3 fields we added, first name, last name and username.

**Pull request merged**